posts/2021-08-13-dev-mode-multi-module.adoc

```diff
@@ -0,0 +1,130 @@
+
+=== Defining the Liberty Maven plug-in
+
+To use the Liberty Maven plug-in in your multi-module project, you can define the plug-in in either the parent `pom.xml` file of every module or in the `pom.xml` of every module.
+
+For example, if your modules have a `<parent>` element that points to a parent project, add the Liberty Maven plug-in to the `<pluginManagement>` section of the parent project's `pom.xml` as follows:
+[source,xml]
+----
+    <build>
+        <pluginManagement>
+            <plugins>
+                <plugin>
+                    <groupId>io.openliberty.tools</groupId>
+                    <artifactId>liberty-maven-plugin</artifactId>
+                    <version>3.4</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+----
+
+=== Starting dev mode
+
+To start dev mode, from the directory containing the top-level multi-module `pom.xml`, run:
+
+----
+mvn liberty:dev
+----
+
+Once dev mode starts successfully, changes in all modules are detected and hot deployed. By default, any dependent modules will also be recompiled and deployed. For example, if a change is made in the `jar` module of the sample application described above, the `war` module will also be recompiled since it depends on the `jar` module. 
+
+**gif of change made to jar module, show compilation of jar and war module**
+
+=== Starting dev mode with container support
+
+There is container support for multiple module projects and dev mode. Use the `devc` goal to start dev mode with container support:
+
+----
+mvn liberty:devc  
+----
+
+If your Dockerfile is not in the same directory as where you started the dev mode command from, you can point to it with the `-Ddockerfile=<path-to-dockerfile>` paremeter. For more information on Dockerfiles for your Open Liberty application, see the link:https://github.com/OpenLiberty/ci.docker#building-an-application-image[Open Liberty Docker image documentation]. 
+
+=== Running tests
+You can run tests on all modules by pressing Enter. 
+
+**gif of enter to run tests**
+
+You can also enable hot tests so that tests will run on every code change. To enable hot tests, add the `-DhotTests` parameter: `mvn liberty:dev -DhotTests`. With hot tests enabled, tests will run on the module that was changed and any dependent modules.
+
+=== Stopping dev mode
+
+You can exit dev mode by pressing Control-C, or type `q` and press Enter in the terminal.
+
+=== Running Liberty in the foreground without dev mode
+
+To run Liberty in the foreground but without dev mode features such as hot deployment, on-demand testing, or debugging, run the `mvn liberty:run` command from the directory containing the top-level multi-module `pom.xml`.  You can stop Liberty by pressing Control-C.
```

Comment:
This is just my impression reading through- what benefit does someone get by running in the foreground? Is it just the ability to launch multiple modules with a single command? That might be worth mentioning when you talk about being able to do that with dev mode but does it need a separate section?  The main benefits of dev mode are not included so it seems odd to highlight it in this post.

Reply:
Yes, it is just the ability to launch multiple modules with a single command, which was not available before.  

I agree we can remove it from this blog post (and just leave it as details within the Liberty Maven plug-in's own release notes etc.).  In this case, should we change the section header `Running Open Liberty on a multi-module project` to be more specific, such as `Running Open Liberty dev mode on a multi-module project`?

Reply:
Yes- that makes sense.

on another note, I was a little confused about how the prepositions generally work here. 

I haven't seen a reference to running dev mode _on a project_. Not saying this is incorrect, just that we should be consistent with our abstractions. Does the project/application run on the server, or in this case do you run OL on the project?

Reply:
The project/application runs on the server, and the lifecycle of both the deployed application and server are managed by the dev goal of the Liberty Maven plug-in.  

So perhaps a heading of `Running a multi-module project with Open Liberty dev mode`?  In that case, would any changes be needed for the sentence `Open Liberty can be run on a multi-module project with a single command by using dev mode or dev mode with container support.`?

Reply:
Based on precedent in the docs, I would expect it to be something like:

`You can run a multi-module project Open Liberty with a single command by using dev mode or dev mode with container support.`

Reply:
Thanks, I've made those changes.